flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/dialect/JdbcDialect.java

```diff
@@ -148,14 +150,15 @@ default String getDeleteStatement(String tableName, String[] conditionFields) {
 	/**
 	 * Get select fields statement by condition fields. Default use SELECT.
 	 */
-	default String getSelectFromStatement(String tableName, String[] selectFields, String[] conditionFields) {
+	default String getSelectFromStatement(String tableName, String[] selectFields, String[] conditionFields, long limit) {
 		String selectExpressions = Arrays.stream(selectFields)
 				.map(this::quoteIdentifier)
 				.collect(Collectors.joining(", "));
 		String fieldExpressions = Arrays.stream(conditionFields)
 				.map(f -> format("%s = :%s", quoteIdentifier(f), f))
 				.collect(Collectors.joining(" AND "));
 		return "SELECT " + selectExpressions + " FROM " +
-				quoteIdentifier(tableName) + (conditionFields.length > 0 ? " WHERE " + fieldExpressions : "");
+				quoteIdentifier(tableName) + (conditionFields.length > 0 ? " WHERE " + fieldExpressions : "") +
+				getLimitStatement(limit);
```

Comment:
Personally, I don't like the current design of the new interface. 
 - it couples the `limit` to the select statement which makes it impossible to inject additional where clause. 
 - dialects need to implement/take care 2 interfaces about limit. 
 - many invokers have to pass an invalid limit. 

In my opinion, a better solution would be separate them, by having a new interface for limit:

```java
String getLimitClause(long limit);
```

And concat the limit caluse with the select statement where needs the limit clause.